crates/trigger/src/key_value.rs

```diff
@@ -0,0 +1,41 @@
+use std::{path::Path, sync::Arc};
+
+use anyhow::{Context, Result};
+use spin_key_value::{CachingStoreManager, DelegatingStoreManager, KeyValueComponent};
+use spin_key_value_sqlite::{DatabaseLocation, KeyValueSqlite};
+
+use crate::runtime_config::RuntimeConfig;
+
+// TODO: Once we have runtime configuration for key-value stores, the user will be able
+// to both change the default store configuration (e.g. use Redis, or an SQLite
+// in-memory database, or use a different path) and add other named stores with their
+// own configurations.
+
+pub(crate) fn build_key_value_component(
+    runtime_config: &RuntimeConfig,
+) -> Result<KeyValueComponent> {
+    let location = match runtime_config.sqlite_db_path() {
+        Some(path) => {
+            // Create the store's parent directory if necessary
+            create_parent_dir(&path).context("Failed to create key value store directory")?;
+            DatabaseLocation::Path(path)
+        }
+        None => DatabaseLocation::InMemory,
+    };
+
+    let manager = Arc::new(CachingStoreManager::new(DelegatingStoreManager::new([(
+        "default".to_owned(),
+        Arc::new(KeyValueSqlite::new(location)) as _,
+    )])));
+
+    Ok(KeyValueComponent::new(spin_key_value::manager(move |_| {
+        manager.clone()
+    })))
+}
+
+fn create_parent_dir(path: &Path) -> Result<()> {
+    let dir = path
+        .parent()
+        .with_context(|| format!("{path:?} missing parent dir"))?;
+    std::fs::create_dir_all(dir).with_context(|| format!("Failed to create parent dir {dir:?}"))
```

Comment:
nit (as if the others weren't also nits): I _think_ that if this fails, it will print "Failed to create key-value store directory" and "Failed to create parent dir ..." - this makes it sound like the second is the parent directory of the first when they are actually the same directory.  Should we maybe drop 'parent' here?  Or tweak the message to something like 'Unable to create K/V store file" (then "parent dir" explains why)?

Reply:
Changed outer context to "Failed to create key value store".